### ISSUE TITLE
Delay - Inconsistent time

### ISSUE THREAD

**Issue by zonkmachine** [LMMS/lmms#3276]
Since 4f76241e6803a61377fe32ef522378d9c5b1b7e5 the timing is off. With an older project it plays back correct but when syncing to project tempo the delay is 4 times too fast. Tempo Sync one whole note and you get a quarter note instead.

```

zonkmachine@zonkmachine:~/builds/lmms/lmms$ git bisect bad
4f76241e6803a61377fe32ef522378d9c5b1b7e5 is the first bad commit
commit 4f76241e6803a61377fe32ef522378d9c5b1b7e5
Author: Dave French <dave.french3@googlemail.com>
Date:   Sat Apr 18 18:40:41 2015 +0100

    Delay and Flanger, removed noise from delay when automating length
    
    The original delay code, was setting it read index relative to the length
    of the delay. This fixes that, now when the length is changed the read
    index stays in the correct place in the buffer.

```

**Comment by zonkmachine** [comment]
Flanger issue could be related: https://github.com/LMMS/lmms/issues/3195

**Comment by zonkmachine** [comment]
> Flanger issue could be related: #3195

:+1:  Yup!

**Comment by zonkmachine** [comment]
Should [m_lfoAmountModel](https://github.com/LMMS/lmms/blob/master/plugins/Delay/DelayControls.cpp#L38) really be a TempoSyncKnobModel? FloatModel is what I would expect.

**Comment by musikBear** [comment]
RC2.8 :
Honestly, i cant make sense of this plugin, (Delay-lmms) 
Rate / LFO creates noise and clips 
I expect synchronizing to a set value (like 1/8) would mean, that there are 1/8 of a whole-note *between* each output.
It could sound like, as @zonkmachine says, it is 4 times faster -at least
-should it be prosponed to 1.2x ?

**Comment by michaelgregorius** [comment]
I have checked this one and here are my observations:
* Please note that commit 4f76241e6803a61377fe32ef522378d9c5b1b7e5 has changed the parameter `max` for `m_delayTimeModel` from 20 seconds to 5 seconds in the constructor of `DelayControls`. This is exactly the factor of 4 that the timings are off.
* If you set the value back to 20 the problem is fixed. However, it seems that a length of 20 seconds was deemed overkill for the delay so a solution should likely fix the problem at the root so that all values for `max` can be used here.
* The conversion to length in seconds is done in `TempoSyncKnobModel::calculateTempoSyncTime`. Here several factors come into play that I cannot make any sense of right now: the magic number of 60000, a scaling factor that's stored in `m_scale` and the maximum value of the model.
* The problem might also be solved by changing the scaling factor in the constructor of `m_delayTimeModel`. However, in case this scaling factor is always tightly coupled with the max value then this should be made more obvious by merging them into one parameter.

Perhaps someone with more knowledge can fix the problem using these pointer.

**Comment by softrabbit** [comment]
60000 = the number of milliseconds in a minute. But apart from that the conversion really _is_ a bit cryptic.

**Comment by jasp00** [comment]
>the conversion really *is* a bit cryptic.

`float oneUnit = 60000.0 / ( _bpm * conversionFactor * m_scale );`

`TempoSyncKnobModel::calculateTempoSyncTime` calculates the period (`oneUnit`) given the beats per minute (`_bpm`). It is a time = frequency⁻¹ formula. First,
Beats per minute / 60000 = Beats per second / 1000 = Beats per millisecond

`conversionFactor` depends on the note length; shorter notes mean more beats. Last, `m_scale` is the maximum period in milliseconds; an initial value would be
`SECS_PER_LFO_OSCILLATION * 1000.0`

**Comment by tresf** [comment]
@zonkmachine do you have time to test this out the proposed fix? #3522
